### PR TITLE
Use GIR bindings instead of pygtk

### PR DESCRIPTION
**Pull request by bigon**

### CODE REVIEW

**Comment by bigon** [comment]
14.04 that's pretty old, you are supporting 3 LTS releases at the same time?

FTR, debian will remove pygtk after buster (so in about 2years) but I think that ubuntu will remove that way earlier (not 100% sure about their planning)

**Comment by nipunn1313** [comment]
Yeah, Dropbox is currently supported back to 14.04.
We periodically deprecate older OSes based on usage numbers.

Does pygtk and GIR bindings both work on the current releases? If so, perhaps we can hold onto this PR until 14.04 is deprecated and land it then.

Otherwise, we'll have to come up with a release strategy to do this (perhaps branching/tagging a build for 14.04 and moving master forward).

**Comment by bigon** [comment]
Yes everything is still working in debian (and in ubuntu too I guess)

On my machine, `nautilus-dropbox` package is the last package depending on `python-gtk2`. But dropbox is [far from being the last package](https://udd.debian.org/cgi-bin/bts-usertags.cgi?tag=pygtk&user=pkg-gnome-maintainers%40lists.alioth.debian.org) using `python-gtk2`

An other solution would be to patch the package in debian, but for such invasive change I personally prefer to have that fixed upstream first

**Comment by augiekim** [comment]
Thanks @bigon for these PRs. We will review them soon then verify if it runs on some OS that we support.

File: dropbox.in [hunk 1]
```diff
@@ -464,15 +468,15 @@ if GUI_AVAILABLE:
             def label_motion(self, widget, event):
                 offx, offy = self.label.get_layout_offsets()
                 layout = self.label.get_layout()
-                index = layout.xy_to_index(int((offx+event.x)*pango.SCALE),
-                                           int((offy+event.y)*pango.SCALE))[0]
+                index = layout.xy_to_index(int((offx+event.x)*Pango.SCALE),
+                                           int((offy+event.y)*Pango.SCALE))[1]
```
[the remark this answers is not in the document]
Comment (bigon) [reply dropbox.in]: Apparently there was a bug in the pygtk binding for this function, the result of `pango_layout_xy_to_index()` C function was not passed/lost by the binding, see https://github.com/GNOME/pygtk/blob/master/pango.override#L1244. so `layout.xy_to_index()` in python return `(index, trailing)`

With the GIR binding, the result of the function is always the 1st member of the tuple, so the function now returns `(inside, index, trailing)`

**Comment by bigon** [comment]
@raveit65 The patch here only modify the `dropbox` executable, not the plugin loaded in nautilus.

So these errors are not related IMVHO

**Comment by raveit65** [comment]
But i don't get this warnings with latest nautilus-dropbox release without the PR.
And there are changes in PR which are GdkPixbuf related.

I'd like to use your PR for rpmfusion for fedora, but i can't spam users terminal with warnings
Fedora 29 released in october is the latest release with python2 and i am in doubt that pygtk will live longer. So you your PR is useful.

**Comment by raveit65** [comment]
Any way it is possible that i have this issue only in fedora 28, need to test it more with fedora 29 or rawhide.
In a quick test with f29 i don't get this warnings, but for some reasons i don't any preview of images in f29 with nautilus.
....nautilus isn't really my favorite application as i am maintaining Mate desktop in fedora ;)

**Comment by raveit65** [comment]
I tested your PR for using with caja-dropbox from MATE desktop. https://github.com/mate-desktop/caja-dropbox ,
and i can't reproduce the problem. So, it looks like that it is indeed a nautilus specific problem.

**Comment by monsta** [comment]
Ubuntu 14.04 has `python-gi` 3.12 (and the packaging is updated for that in https://github.com/dropbox/nautilus-dropbox/pull/67), so I guess you can remove `threads_init`...

**Pull request merged**